lib/java/src/org/apache/thrift/server/TThreadPoolServer.java

```diff
@@ -343,5 +341,10 @@ public void run() {
         }
       }
     }
+
+    private boolean isIgnorableTSaslTransportException(Exception x) {
+      return (x instanceof TSaslTransportException && ERR_NO_SASL_DATA_IN_STREAM.equals(x.getMessage()))
```

Comment:
Relying on parsing an exception's message is less than ideal.

Reply:
I'm with you, sir. I could just remove that part because, at the moment, `TSaslTransportException` is thrown only in the `END_OF_FILE` case.
In other words, if `(x instanceof TSaslTransportException)`, then `x.getMessage()` is always `ERR_NO_SASL_DATA_IN_STREAM`.
I'll update appropriately.

Reply:
I'd say get rid of TSaslTransportException all together.  I don't see a need for it?

Reply:
You're right. I'll simplify this.

Reply:
Removed `TSaslTransportException`. Added direct checks for `END_OF_FILE`.